src/backend/api/Fusion.Resources.Domain/Services/ProfileServices.cs

```diff
@@ -266,6 +266,23 @@ public async Task<DbExternalPersonnelPerson> EnsureExternalPersonnelAsync(string
             }
         }
 
+        public Task<IEnumerable<ResolvedPersonProfile>?> ResolveProfilesAsync(IEnumerable<PersonId> personIds)
+        {
+            return ResolveProfilesAsync(personIds.Select(x => (PersonIdentifier)x));
+        }
+
+        public async Task<IEnumerable<ResolvedPersonProfile>?> ResolveProfilesAsync(IEnumerable<PersonIdentifier> personIds)
+        {
+            try
+            {
+                return await profileResolver.ResolvePersonsAsync(personIds);
```

Comment:
Not sure if relevant here, but recently observed there is a limit of 500 items in list here.